include/vfio-user.h

```diff
@@ -114,18 +114,27 @@ struct vfio_user_bitmap {
     char data[];
 } __attribute__((packed));
 
-struct vfio_user_dma_region {
-    uint64_t                addr;
-    uint64_t                size;
-    uint64_t                offset;
-    uint32_t                prot;
+/* based on strcut vfio_iommu_type1_dma_map */
+struct vfio_user_dma_map {
+    uint32_t argsz;
+#define VFIO_USER_F_DMA_REGION_MAPPABLE (1 << 2)
```

Comment:
Can we just have VFIO_USER_F_DMA_REGION_READ/WRITE too please? It's a bit weird to mix-and-match IMO

Reply:
sure